azure_auth/handlers.py

```diff
@@ -92,6 +92,8 @@ def authenticate(self, token: dict) -> UserModel:
         try:
             user = UserModel._default_manager.get_by_natural_key(email)
         except UserModel.DoesNotExist:
+            if not settings.AZURE_AUTH.get("CREATE_NEW_USER", True):
```

Comment:
Will need a `self.request.session.flush()` here to clear the id_token claims from the session I reckon.